In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

# 1. Đọc bộ dữ liệu Credit Card
# (Lưu ý: Đổi tên file 'default of credit card clients.xls' cho đúng với tên file tải về)
df = pd.read_excel('default of credit card clients.xls', header=1)

# 2. Phân chia dữ liệu: X (thông tin đầu vào) và y (mục tiêu: có vỡ nợ hay không)
X = df.drop(columns=['ID', 'default payment next month'])
y = df['default payment next month']

# 3. Cắt 20% dữ liệu ra để làm bài thi thử (test)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 4. Huấn luyện mô hình phân loại (Random Forest)
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

# 5. Đánh giá kết quả trên tập thi thử
y_pred = model.predict(X_test)
do_chinh_xac = accuracy_score(y_test, y_pred) * 100
print(f"Độ chính xác của mô hình phát hiện vỡ nợ tín dụng là: {do_chinh_xac:.2f}%")

Độ chính xác của mô hình phát hiện vỡ nợ tín dụng là: 81.60%


Mô hình Random Forest đã phân tích 30,000 khách hàng từ bộ dữ liệu Credit Card.
Độ chính xác đạt 81.60%, cho thấy mô hình có khả năng nhận diện tốt các đặc điểm của khách hàng có nguy cơ nợ xấu dựa trên lịch sử thanh toán và dư nợ.